src/validator.cpp

```diff
@@ -980,22 +980,32 @@ void Validator::ValidatorImpl::validateConnections(const ModelPtr &model)
                             if (!unitsAreEquivalent(model, variable, equivalentVariable, hints, multiplier)) {
                                 auto unitsName = variable->units() == nullptr ? "" : variable->units()->name();
                                 auto equivalentUnitsName = equivalentVariable->units() == nullptr ? "" : equivalentVariable->units()->name();
-                                ErrorPtr err = Error::create();
-                                err->setDescription("Variable '" + variable->name() + "' has units of '" + unitsName + "' and an equivalent variable '" + equivalentVariable->name() + "' with non-matching units of '" + equivalentUnitsName + "'. The mismatch is: " + hints);
-                                err->setModel(model);
-                                err->setKind(Error::Kind::UNITS);
-                                mValidator->addError(err);
+                                IssuePtr issue = Issue::create();
+                                issue->setDescription("Variable '" + variable->name() + "' has units of '" + unitsName + "' and an equivalent variable '" + equivalentVariable->name() + "' with non-matching units of '" + equivalentUnitsName + "'. The mismatch is: " + hints);
+                                issue->setModel(model);
+                                issue->setCause(Issue::Cause::UNITS);
+                                mValidator->addIssue(issue);
+                            } else if (multiplier != 0.0) {
+                                // Warning when the multipliers are not the same.
+                                auto unitsName = variable->units() == nullptr ? "" : variable->units()->name();
+                                auto equivalentUnitsName = equivalentVariable->units() == nullptr ? "" : equivalentVariable->units()->name();
+                                IssuePtr issue = Issue::create();
+                                issue->setDescription("Variable '" + variable->name() + "' has units of '" + unitsName + "' and an equivalent variable '" + equivalentVariable->name() + "' with non-matching units of '" + equivalentUnitsName + "'. The mismatch is: " + hints);
+                                issue->setModel(model);
+                                issue->setLevel(libcellml::Issue::Level::WARNING);
```

Comment:
How does the validator create warnings, it either breaks a rule of the specification or it doesn't.

Reply:
This is the only occurrence of the `WARNING` use, just to demonstrate how it will be used in the future.  In this place in the validator it's used to return a warning when the multipliers are mismatched, but the base units are consistent.

Reply:
This isn't about the future it is about the now.  Now the validator doesn't (nor in my opinion ever should) report warnings.

Reply:
The user doesn't need to get them if they don't want them, because as it stands all the current functions which use errorCount() and error(index) will behave identically to how they used to.  
... but if/when they do, the warnings and hints will be available.  

It's a whole lot more clumsy to have a separate class or tool to create warnings when the validator is actually doing all the checking already, just not reporting it.  This is what we spoke about last month? That the strict validation against the specifications only would be a subset of the functionality of the checker/validator/whatever-we-call it?

Reply:
We spoke about this month but we can't have warnings in the validator.  The validator checks a model against the specification reporting violations of the spec. as errors.  In this instance I think it is a good idea to ignore what is going on under the hood and just focusing on the interface that is presented to users.

Reply:
OK, I'm really confused now!  I'll remove it so that this PR can go through, but I'm struggling to understand how this will work nicely ... :|